### DISCUSSION TITLE
v6.1.0 populate doesn't  inherit to toArray,toItems() loaded properties

### DISCUSSION THREAD

**Discussion by tkudlicka** [mikro-orm/mikro-orm#5227] [General]
### Describe the bug

When i use em.populate(items,['deepneste.property.in.collection'] options as collection to v6.1.0 and then try to load only specific keys and then convert to array, it doesn't inherit Loaded keys to specific entity of collection.  
it also same problem for `getItems`

### Reproduction

```
import { Collection, Loaded, ManyToOne, OneToMany, Property } from '@mikro-orm/core';
import { Ref } from '@mikro-orm/core/entity';

class Car {
  @Property()
  test: string;

  @OneToMany(() => Engine,x => x.car)
  engines = new Collection<Engine>(this);
}

export class Engine {
  @Property()
  name: string;


  @ManyToOne(() => Car, {
    nullable: false,
    ref: true
  })
  car: Ref<Car>;
 
  @ManyToOne(() => OilPan, {
    nullable: false,
    ref: true
  })
  oilPan?: Ref<OilPan>;
}

export class OilPan {
  @Property()
  name: string;

  @OneToMany(() => Engine,x => x.oilPan)
  engines = new Collection<Engine>(this);
}
```
and then i use populate for items
```
[const oilPans =await this.mtem.findAll(OilPan,{
      populate: ['engines.cars.test']
    })
    const array = oilPans[0].engines.toArray() 

```

### What driver are you using?

Postgres

### MikroORM version

v6.1.0

### Node.js version

18

### Operating system

mac Sonoma 14.1.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I am not sure what you are reporting here, maybe try to provide _complete_ reproduction so it's clear what you are doing and what is not working for you.

> const cars =await this.mtem.populate(oilPans,{
>      populate: ['engines.cars.name']
>    })

This looks wrong, you don't get "cars" from this call, you get the exact same thing as you put inside - the `oilPans` entities (whatever type they are).

>    const array = collection.toArray()

And here I have no idea what is `collection`.

**Reply by tkudlicka** [reply]
array populated with deep relationship are not typed correctly as the return type is plan entity of collection and it should have  loaded wrapper with the populated fields.. i upgraded the example a little
![Screenshot 2024-02-09 at 13 41 46](https://github.com/mikro-orm/mikro-orm/assets/70945365/d03bab5a-6b52-4307-9b62-be415e4dfe5c)
![Screenshot 2024-02-09 at 13 42 26](https://github.com/mikro-orm/mikro-orm/assets/70945365/50808344-058a-42c8-924f-e75f5fab3b3c)
it should correctly be ```Loaded<Engine, "oilPan.name", "*", never>```

**Reply by B4nan** [reply]
they are typed correctly on my end  ¯\_(ツ)_/¯ 

so again, please provide a complete reproduction so it's clear what you are doing.

**Reply by tkudlicka** [reply]
The issue is in entity oilPan for Collection<Engine>(this), it creates a new instance of Collection that doesn't automatically inherit loaded properties.
```
import { Collection,Ref, ManyToOne, OneToMany, Property } from '@mikro-orm/core';
import { MikroORM } from '@mikro-orm/postgresql';
class Car {
  @Property()
  test: string;

  @OneToMany(() => Engine,x => x.car)
  engines = new Collection<Engine>(this);
}

export class Engine {
  @Property()
  name: string;


  @ManyToOne(() => Car, {
    nullable: false,
    ref: true
  })
  car: Ref<Car>;
 
  @ManyToOne(() => OilPan, {
    nullable: false,
    ref: true
  })
  oilPan?: Ref<OilPan>;
}

export class OilPan {
  @Property()
  name: string;

  @OneToMany(() => Engine,x => x.oilPan)
  engines = new Collection<Engine>(this);
}


async function test () {
    const {em} = MikroORM.initSync({} as any);
    const oilPans =await em.findAll(Car,{
        populate: ['engines.oilPan.name']
      })
      for (const pan of oilPans) {
        const item = pan.engines.getItems()
      }
}

```

**Reply by B4nan** [reply]
And what is not working for you there? The code compiles just fine on my end.

**Reply by B4nan** [reply]
Note that including the `name` in your populate hint has no effect. Populating is not partial loading.

**Reply by tkudlicka** [reply]
i should be able to get property name without loading entity right?
`item[0].oilPan?.name` and it doesnt include loaded entity

**Reply by B4nan** [reply]
I see, you need to use the `$` property which will give you the loaded collection instance:

<img width="597" alt="image" src="https://github.com/mikro-orm/mikro-orm/assets/615580/c1055d85-b85d-4588-a2db-90f44fc9d45b">

**Reply by tkudlicka** [reply]
oh okay, thanks
